optuna/storages/grpc/api.proto

```diff
@@ -0,0 +1,430 @@
+ */
+message GetStudyUserAttributesReply {
+    map<string, string> user_attributes = 1;
+}
+
+/**
+ * Request to get study system attributes.
+ */
+message GetStudySystemAttributesRequest {
+    int64 study_id = 1;
+}
+
+/**
+ * Reply to get study system attributes.
+ */
+message GetStudySystemAttributesReply {
+    map<string, string> system_attributes = 1;
+}
+
+/**
+ * Request to get all studies.
+ */
+message GetAllStudiesRequest {}
+
+/**
+ * Reply to get all studies.
+ */
+message GetAllStudiesReply {
+    repeated FrozenStudy frozen_studies = 1;
+}
+
+/**
+ * Request to create a new trial.
+ */
+message CreateNewTrialRequest {
+    int64 study_id = 1;
+    FrozenTrial template_trial = 2;
+    bool template_trial_is_none = 3;
+}
+
+/**
+ * Reply to create a new trial.
+ */
+message CreateNewTrialReply {
+    int64 trial_id = 1;
+}
+
+/**
+ * Request to set a trial parameter.
+ */
+message SetTrialParameterRequest {
+    int64 trial_id = 1;
+    string param_name = 2;
+    double param_value_internal = 3;
+    string distribution = 4;
+}
+
+/**
+ * Reply to set a trial parameter.
+ */
+message SetTrialParameterReply {}
+
+/**
+ * Request to get a trial id from its study id and trial number.
+ */
+message GetTrialIdFromStudyIdTrialNumberRequest {
+    int64 study_id = 1;
+    int64 trial_number = 2;
+}
+
+/**
+ * Reply to get a trial id from its study id and trial number.
+ */
+message GetTrialIdFromStudyIdTrialNumberReply {
+    int64 trial_id = 1;
+}
+
+/**
+ * Request to set trial state and values.
+ */
+message SetTrialStateValuesRequest {
+    int64 trial_id = 1;
+    TrialState state = 2;
+    repeated double values = 3;
+}
+
+/**
+ * Reply to set trial state and values.
+ */
+message SetTrialStateValuesReply {
+    bool trial_updated = 1;
+}
+
+/**
+ * Request to set a trial intermediate value.
+ */
+message SetTrialIntermediateValueRequest {
+    int64 trial_id = 1;
+    int64 step = 2;
+    double intermediate_value = 3;
+}
+
+/**
+ * Reply to set a trial intermediate value.
+ */
+message SetTrialIntermediateValueReply {}
+
+/**
+ * Request to set a trial user attribute.
+ */
+message SetTrialUserAttributeRequest {
+    int64 trial_id = 1;
+    string key = 2;
+    string value = 3;
+}
+
+/**
+ * Reply to set a trial user attribute.
+ */
+message SetTrialUserAttributeReply {}
+
+/**
+ * Request to set a trial system attribute.
+ */
+message SetTrialSystemAttributeRequest {
+    int64 trial_id = 1;
+    string key = 2;
+    string value = 3;
+}
+
+/**
+ * Reply to set a trial system attribute.
+ */
+message SetTrialSystemAttributeReply {}
+
+/**
+ * Request to get a trial by its ID.
+ */
+message GetTrialRequest {
+    int64 trial_id = 1;
+}
+
+/**
+ * Reply to get a trial by its ID.
+ */
+message GetTrialReply {
+    FrozenTrial frozen_trial = 1;
+}
+
+/**
+ * Request to get all trials in a study.
+ */
+message GetAllTrialsRequest {
+    int64 study_id = 1;
+    bool deepcopy = 2;
+    repeated TrialState states = 3;
+}
```

Comment:
```suggestion
message GetAllTrialsRequest {
    int64 study_id = 1;
    repeated TrialState states = 2;
}
```

This option is not required since all frozen trials are forced to be copied even when specifies `deepcopy=false`